StringEncryption.cpp

```diff
@@ -175,48 +193,32 @@ struct StringEncryption : public ModulePass {
           *(GV->getParent()), EncryptedConst->getType(), false,
           GV->getLinkage(), EncryptedConst, "EncryptedString", nullptr,
           GV->getThreadLocalMode(), GV->getType()->getAddressSpace());
-      old2new[GV] = EncryptedRawGV;
-      GV2Keys[EncryptedRawGV] = KeyConst;
+      GlobalVariable *DecryptSpaceGV = new GlobalVariable(
+          *(GV->getParent()), DummyConst->getType(), false,
+          GV->getLinkage(), DummyConst, "DecryptSpace", nullptr,
+          GV->getThreadLocalMode(), GV->getType()->getAddressSpace());
+      old2new[GV] = make_pair(EncryptedRawGV, DecryptSpaceGV);
+      GV2Keys[DecryptSpaceGV] = make_pair(KeyConst, EncryptedRawGV);
     }
     // Now prepare ObjC new GV
     for (GlobalVariable *GV : objCStrings) {
-      Value *zero = ConstantInt::get(Type::getInt32Ty(GV->getContext()), 0);
-      ConstantStruct *CS = cast<ConstantStruct>(GV->getInitializer());
-      vector<Constant *> vals;
-      vals.push_back(CS->getOperand(0));
-      vals.push_back(CS->getOperand(1));
       GlobalVariable *oldrawString =
-          cast<GlobalVariable>(CS->getOperand(2)->stripPointerCasts());
-      if (old2new.find(oldrawString) ==
-          old2new.end()) { // Filter out zero initializers
+          cast<GlobalVariable>(cast<ConstantStruct>(GV->getInitializer())->getOperand(2)->stripPointerCasts());
```

Comment:
``cast<ConstantStruct>(GV->getInitializer())`` is already assigned to ``CS``, try avoiding duplicate function call here